specification/storagetasks/resource-manager/Microsoft.StorageTasks/stable/2023-01-01/storagetasks.json

```diff
@@ -0,0 +1,903 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}": {
+      "put": {
+        "tags": [
+          "StorageTasks"
+        ],
+        "description": "Asynchronously creates a new storage task resource with the specified parameters. If a storage task is already created and a subsequent create request is issued with different properties, the storage task properties will be updated. If a storage task is already created and a subsequent create or update request is issued with the exact same set of properties, the request will succeed.",
+        "operationId": "StorageTasks_Create",
+        "x-ms-examples": {
+          "PutStorageTask": {
+            "$ref": "./examples/storageTasksCrud/PutStorageTask.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/StorageTaskName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StorageTask"
+            },
+            "description": "The parameters to create a Storage Task."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Update storage task with the same properties specified in the request body.",
+            "schema": {
+              "$ref": "#/definitions/StorageTask"
+            }
+          },
+          "201": {
+            "description": "OK -- Create storage task with the same properties specified in the request body.",
+            "schema": {
+              "$ref": "#/definitions/StorageTask"
+            }
+          },
+          "202": {
```

Comment:
>202 [](http://example.com/codeflow?start=11&length=3)

Put shouldn't have 202 response, use 201 for async long running response.

Reply:
According to guideline. PUT on create will be 201, PUT/PATCH on update with be 200
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md

Reply:
Refer to ARM RPC: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously for ARM specific guidance.

Reply:
Quoting from the document 
"Some of the resource provider's existing PUT APIs return 202 Accepted. Please note this is a old model. For PUT requests, 201 / 200 + ProvisioningState is the preferred model. "

The preferred model looks has changed from 202 -> 201 to 201 with a provisioning state. However we have been following 202 to 201 for creates and 202 to 200 for updates for existing Microsoft.Storage type and Microsoft.StorageTasks is a component of Microsoft.Storage and resuses the same code/framework and the implementation in prod regions for testing and integration purposes seem to be working well with ARM templates and portal Bicep templates